framework/SupervisedLearning/DynamicModeDecompositionControl.py

```diff
@@ -384,7 +400,18 @@ def _evaluateMatrices(self, X1, X2, U, Y1, rankSVD):
     # Omega Matrix, stack X1 and U
     omega = np.concatenate((X1, U), axis=0)
     # SVD
-    Ut, St, Vt = mathUtils.computeTruncatedSingularValueDecomposition(omega, rankSVD, True, False)
+    Utsvd, stsvd, Vtsvd = mathUtils.computeTruncatedSingularValueDecomposition(omega, rankSVD, False, False)
+    # Find the truncation rank triggered by "s>=SminValue"
+    rank_s = sum(map(lambda x : x>=1e-6, stsvd.tolist()))
```

Comment:
this should be internalized as an additional option in ```computeTruncatedSingularValueDecomposition```

Reply:
I am afraid that, building it into `computeTruncatedSingularValueDecomposition ` will cause problems to regular DMD function. I found that DMDc will lose its robustness (return different results on different computers) when the minimum value of `St` matrices gets too small, so I performed a external truncation `rank_s` to remove the small values.